crates/ruff_server/src/session/settings.rs

```diff
@@ -234,7 +235,8 @@ impl ResolvedClientSettings {
                     settings
                         .configuration
                         .as_ref()
-                        .map(|config_path| OsString::from(config_path.clone()).into())
+                        .and_then(|config_path| shellexpand::full(config_path).ok())
+                        .map(|config_path| PathBuf::from(config_path.as_ref()))
```

Comment:
Is there a specific reason that we needed `OsString::from` then `into()` here before? Want to make sure I'm not breaking anything.

Reply:
No, that was just a conversion from `String` to `PathBuf` 🙂